### PR TITLE
grub-mkconfig: Do not allow grub.conf change on ostree systems

### PR DESCRIPTION
**Pull request by carlocaione**
https://phabricator.endlessm.com/T19614

### CODE REVIEW

File: util/grub-mkconfig.in [hunk 1]
```diff
@@ -130,6 +130,11 @@ if [ "$EUID" != 0 ] ; then
   fi
 fi
 
+if [ -L /ostree ] ; then
+  gettext_printf "%s: You cannot change grub configuration on as ostree system\n" "$self" >&2
```
Comment (wjt) [inline util/grub-mkconfig.in]: Will this work when `update-grub` is run from `eos-convert-system`? I believe `/ostree` will still be a symlink in that case?

**Comment by dbnicholson** [comment]
The more proper way to detect if you're running an ostree system is to look for `ostree=*` in `/proc/cmdline`.

File: util/grub-mkconfig.in (hunk 1 above)
Comment (carlocaione) [reply util/grub-mkconfig.in]: good point. At this point probably we want to have a new (hidden?) parameter to force it anyway.

File: util/grub-mkconfig.in [hunk 2]
```diff
@@ -130,6 +133,11 @@ if [ "$EUID" != 0 ] ; then
   fi
 fi
 
+if fgrep -qs 'ostree' "/proc/cmdline" && [ "$force" != true ] ; then
```
Comment (dbnicholson) [inline util/grub-mkconfig.in]: I think the grep should be for `ostree=` so you don't match on some other random command line parameter that happens to contain `ostree`.

Comment (carlocaione) [reply util/grub-mkconfig.in]: @dbnicholson fixed that

File: util/grub-mkconfig.in [hunk 3]
```diff
@@ -130,6 +133,11 @@ if [ "$EUID" != 0 ] ; then
   fi
 fi
 
+if fgrep -qs 'ostree=' "/proc/cmdline" && [ "$force" != true ] ; then
+  gettext_printf "%s: You cannot change grub configuration on as ostree system\n" "$self" >&2
```
Comment (wjt) [inline util/grub-mkconfig.in]: Typo: "on as ostree" should be "on an ostree". I'd capitalise GRUB as well.

I wonder what Fedora Atomic Workstation does for turning BLS entries into GRUB menu entries? Do they carry a patch like this?

Comment (carlocaione) [reply util/grub-mkconfig.in]: Fixed the typo and moved the discussion to phabricator.

**Pull request merged**